sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -721,6 +720,11 @@ abstract class RapidsShuffleThreadedReaderBase[K, C](
         shuffleReadTimeNs.foreach(_ += shuffleReadTime)
         dataReadSize.foreach(_ += uncompressedSize)
 
+        // Reset fetch, readBlocked, and wait times to prevent extraneous accruals
+        waitTime = 0L
```

Comment:
Rather than reset this here, I think a better fix is to remove `waitTime` as a field of this class and instead make it a local variable of this method, initialized to zero.

Reply:
Changed in latest commit to have accumulators update when sub-metric accruals happen